File: backend/app/models/section.py
"""카드 섹션 관련 Pydantic 모델"""

from pydantic import BaseModel
from typing import Optional, Literal, Dict


class CardSection(BaseModel):
    """카드 섹션 모델"""
    id: Optional[str] = None
    project_id: str
    order: int
    type: Literal['title', 'content', 'closing']
    title: Optional[str] = None
    content: str
    image_url: Optional[str] = None
    design_config: Dict = {
        'background_color': '#FFFFFF',
        'font_family': 'Pretendard',
        'font_size': 16
    }


class SectionCreate(BaseModel):
    """섹션 생성 요청"""
    order: int
    type: Literal['title', 'content', 'closing']
    title: Optional[str] = None
    content: str


class SectionUpdate(BaseModel):
    """섹션 수정 요청"""
    title: Optional[str] = None
    content: Optional[str] = None
    order: Optional[int] = None


class SectionResponse(BaseModel):
    """섹션 응답"""
    id: str
    project_id: str
    order: int
    type: Literal['title', 'content', 'closing']
    title: Optional[str] = None
    content: str
    design_config: Dict

File: frontend/lib/types/section.ts
/**
 * 카드 섹션 관련 TypeScript 타입 정의
 */

export type CardType = 'title' | 'content' | 'closing';

export interface DesignConfig {
  background_color: string;
  font_family: string;
  font_size: number;
}

export interface CardSection {
  id: string;
  project_id: string;
  order: number;
  type: CardType;
  title?: string;
  content: string;
  design_config: DesignConfig;
}

export interface SectionUpdate {
  title?: string;
  content?: string;
  order?: number;
}

